src/System.Text.Json/tests/Utf8JsonReaderTests.cs

```diff
@@ -3333,5 +3333,57 @@ public static IEnumerable<object[]> CommentTestLineSeparators
                 };
             }
         }
+
+        public static IEnumerable<object[]> GetCommentTestData
+        {
+            get
+            {
+                var dataList = new List<object[]>();
+                foreach (var delim in new[] { "\r", "\r\n", "\n" })
+                {
+                    // NOTE: Leading and trailing spaces in the comments are significant.
+                    var singleLineComment = " Single Line Comment ";
+                    dataList.Add(new object[] { $"{{//{singleLineComment}{delim}}}", singleLineComment });
+
+                    var multilineComment = $" Multiline {delim} Comment ";
+                    dataList.Add(new object[] { $"{{/*{multilineComment}*/{delim}}}", multilineComment });
+                }
+                return dataList;
+            }
+        }
+
+        public static IEnumerable<object[]> GetCommentUnescapeData
+        {
+            get
+            {
+                var dataList = new List<object[]>();
+
+                var rawComments = new string[]
+                {
+                    "A string with {0}valid UTF8 \\t tab",
+                    "A string with {0}invalid UTF8 \\xc3\\x28",
+                    "A string with {0}valid UTF16 \\u002e \\u0009 म",
+                    "A string with {0}invalid UTF16 \\uDD1E"
+                };
+
+                // single line comments
+                foreach (var raw in rawComments)
+                {
+                    var str = string.Format(raw, "");
+                    var cmt = "//" + str;
+                    dataList.Add(new object[] { cmt, str });
+                }
+
+                // multiline comments
+                foreach (var raw in rawComments)
+                {
+                    var str = string.Format(raw, "\n");
```

Comment:
Same here/and below. Avoid using var.

Reply:
hmm... you mean `var` at line 3378 or even in 3380? Because in latter place, the RHS gives sufficient information that a string is being assigned...

Reply:
Both.

3378 gives no clue to the reviewer what `raw` would be at all without scrolling back to 3361)

3380 is "clearly" a string because everyone knows what `string.Format` does.  But the style guidelines in corefx are effectively that `var` can only be used with `new`, `as`, or a hard-cast -- the name of the method being invoked is not sufficient. The fact that it's a static method on `string` doesn't convey the return type, `string.IsNullOrEmpty(raw)` is a boolean.

Reply:
@bartonjs thanks for the clarification. It helps.